src/history/HistoryTests.cpp

```diff
@@ -133,6 +172,25 @@ dbModeName(Config::TestDbMode mode)
     }
 }
 
+static void
+downloadStatusCheck(Work& parent, bool success)
+{
+    for (auto const& child : parent.getChildren())
+    {
+        if (child.second->isDone())
```

Comment:
I think that in the context of tests (that should be deterministic):
* we should have `REQUIRE(child.second->isDone())` (otherwise, we just silently skip work)
* on the actual state of `child.second`,
   * we should not check what children of children are doing (I am not sure why we do this?)
   * instead we should check that `child.second->getState()` has the expected value

To make things clearer, we could have this function take a pointer to `DownloadBucketsWork` instead of just `Work`.